lib/endpoints/class-wp-rest-users-controller.php

```diff
@@ -92,6 +91,10 @@ public function get_items( $request ) {
 		$prepared_args['orderby'] = $orderby_possibles[ $request['orderby'] ];
 		$prepared_args['search'] = $request['search'];
 
+		if ( ! current_user_can( 'list_users' ) ) {
+			$prepared_args['has_published_posts'] = true;
```

Comment:
The thought was this would only reflect users who were already exposed via the RSS feeds, which could be any public post type, though I guess that's not a guarantee either. `show_in_rest` might be safer, or literally `post_type=post`